tests/framework/extensions/workloads/pods/pod_status.go

```diff
@@ -78,3 +67,36 @@ func StatusPods(client *rancher.Client, clusterID string) ([]string, []error) {
 
 	return podResults, podErrors
 }
+
+// CheckPodStatus is a helper function that uses the steve client to check the status of a single pod
+func CheckPodStatus(pod *v1.SteveAPIObject) (podResults string, podError error, err error) {
```

Comment:
There are some fixes for this function that seem like they are already in this PR's history. Could we add the fixes to this function back? https://github.com/rancher/rancher/pull/42368

Reply:
this is added in the 'checkPodStatus' function, but I will make sure it matches more closely now.